addon/run-task.js

```diff
@@ -114,8 +114,17 @@ export function scheduleTask(obj, queueName, taskOrName, ...args) {
   );
 
   let task = getTask(obj, taskOrName, 'scheduleTask');
-  let cancelId = run.schedule(queueName, obj, task, ...args);
   let timers = getTimers(obj);
+  let cancelId;
+  let taskWrapper = (...taskArgs) => {
+    // clean up
+    let index = timers.indexOf(cancelId);
```

Comment:
It seems like there may be a bit of duplication between `runTask` and this. Could we just toss that into a function?